sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -281,11 +281,9 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         trigger = trigger)
     } else {
       val ds = DataSource.lookupDataSource(source, df.sparkSession.sessionState.conf)
-      val sink = (ds.newInstance(), trigger) match {
-        case (w: ContinuousWriteSupport, _: ContinuousTrigger) => w
-        case (_, _: ContinuousTrigger) => throw new UnsupportedOperationException(
-            s"Data source $source does not support continuous writing")
-        case (w: MicroBatchWriteSupport, _) => w
+      val disabledSources = df.sparkSession.sqlContext.conf.disabledV2StreamingWriters.split(",")
```

Comment:
is this option created for data sources that implement both v1 and v2 APIs?

Reply:
Yes. The intent is to support two things:

* Turning off the V2 implementation for a source if someone has problems with it.
* Running some subset of Spark unit tests with V2 disabled, ensuring that the V1 execution pipeline still works for as long as we support it.

Reply:
ok so this is only useful for built-in stream sources, as the v1 source API is not public,